test/StrategyFactory.integration.t.sol

```diff
@@ -0,0 +1,222 @@
+
+    // Addresses
+    address public admin;
+    address public backend;
+    address public guardian;
+    address public deployer;
+    address public mamoMultisig;
+
+    // Strategy parameters
+    uint256 public strategyTypeId;
+    uint256 public splitMToken;
+    uint256 public splitVault;
+
+    function setUp() public {
+        // Create a new addresses instance for testing
+        string memory addressesFolderPath = "./addresses";
+        uint256[] memory chainIds = new uint256[](1);
+        chainIds[0] = block.chainid;
+
+        addresses = new Addresses(addressesFolderPath, chainIds);
+
+        // Get the environment from command line arguments or use default
+        string memory environment = vm.envOr("DEPLOY_ENV", string("8453_PROD"));
+        string memory configPath = string(abi.encodePacked("./deploy/", environment, ".json"));
+        string memory assetConfigPath = vm.envString("ASSET_CONFIG_PATH");
+
+        // Load configurations
+        DeployConfig configDeploy = new DeployConfig(configPath);
+        config = configDeploy.getConfig();
+
+        DeployAssetConfig assetConfigDeploy = new DeployAssetConfig(assetConfigPath);
+        assetConfig = assetConfigDeploy.getConfig();
+
+        // Get the addresses for the roles
+        admin = addresses.getAddress(config.admin);
+        backend = addresses.getAddress(config.backend);
+        guardian = addresses.getAddress(config.guardian);
+        deployer = addresses.getAddress(config.deployer);
+        mamoMultisig = admin; // Use admin as mamo multisig for testing
+
+        StrategyFactoryDeployer factoryDeployer = new StrategyFactoryDeployer();
+        address factoryAddress = factoryDeployer.deployStrategyFactory(addresses, assetConfig);
+        factory = StrategyFactory(payable(factoryAddress));
+
+        registry = MamoStrategyRegistry(addresses.getAddress("MAMO_STRATEGY_REGISTRY"));
+
+        vm.startPrank(mamoMultisig);
+        registry.grantRole(registry.BACKEND_ROLE(), address(factory));
+        vm.stopPrank();
+
+        splitMToken = assetConfig.strategyParams.splitMToken;
+        splitVault = assetConfig.strategyParams.splitVault;
+        strategyTypeId = assetConfig.strategyParams.strategyTypeId;
+    }
+
+    function testFactoryDeployment() public view {
+        // Test that the factory was deployed correctly
+        assertTrue(address(factory) != address(0), "Factory not deployed");
+
+        // Test that the factory has the correct parameters
+        assertEq(factory.mamoStrategyRegistry(), address(registry), "Registry address mismatch");
+        assertEq(factory.mamoBackend(), backend, "Backend address mismatch");
+        assertEq(factory.splitMToken(), splitMToken, "Split mToken mismatch");
+        assertEq(factory.splitVault(), splitVault, "Split vault mismatch");
+        assertEq(factory.strategyTypeId(), strategyTypeId, "Strategy type ID mismatch");
+    }
+
+    function testCreateStrategyForUser() public {
```

Comment:
please also add a test that a non-backend wallet can't create on behalf of a user